logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/receiver/UserEventReceiver.kt

```diff
@@ -31,13 +33,20 @@ class UserEventReceiverImpl(
 
     private suspend fun handleClientRemove(event: Event.User.ClientRemove) {
         clientRepository.currentClientId().map { currentClientId ->
-            if (currentClientId.value == event.id)
+            if (currentClientId == event.clientId)
                 logoutUseCase(LogoutReason.REMOVED_CLIENT)
         }
     }
 
     private suspend fun handleUserDelete(event: Event.User.UserDelete) {
-        logoutUseCase(LogoutReason.DELETED_ACCOUNT)
+        sessionRepository.currentSession().map { currentSession ->
+            if (currentSession.session.userId == event.userId) {
+                logoutUseCase(LogoutReason.DELETED_ACCOUNT)
+            } else {
+                /* TODO: handle a connection delete their account:
```

Comment:
we need to add a task to handle this case

Reply:
yes, I'll add it to the backlog.